msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/util/ByteBuffer/BooleanArrays.java

```diff
@@ -0,0 +1,844 @@
+/* Generic definitions */
+/* Assertions (useful to generate conditional code) */
+/* Current type and class (and size, if applicable) */
+/* Value methods */
+/* Interfaces (keys) */
+/* Interfaces (values) */
+/* Abstract implementations (keys) */
+/* Abstract implementations (values) */
+/* Static containers (keys) */
+/* Static containers (values) */
+/* Implementations */
+/* Synchronized wrappers */
+/* Unmodifiable wrappers */
+/* Other wrappers */
+/* Methods (keys) */
+/* Methods (values) */
+/* Methods (keys/values) */
+/* Methods that have special names depending on keys (but the special names depend on values) */
+/* Equality */
+/* Object/Reference-only definitions (keys) */
+/* Primitive-type-only definitions (keys) */
+/* Object/Reference-only definitions (values) */
+/*
+ * Copyright (C) 2002-2014 Sebastiano Vigna
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ *
+ *
+ * For the sorting and binary search code:
+ *
+ * Copyright (C) 1999 CERN - European Organization for Nuclear Research.
+ *
+ * Permission to use, copy, modify, distribute and sell this software and its documentation for any
+ * purpose is hereby granted without fee, provided that the above copyright notice appear in all
+ * copies and that both that copyright notice and this permission notice appear in supporting
+ * documentation. CERN makes no representations about the suitability of this software for any
+ * purpose. It is provided "as is" without expressed or implied warranty.
+ */
+package io.github.msdk.io.mzml2.util.ByteBuffer;
+
+import java.util.Random;
+
+/**
+ * A class providing static methods and objects that do useful things with type-specific arrays.
+ *
+ * <p>
+ * In particular, the <code>ensureCapacity()</code>, <code>grow()</code>, <code>trim()</code> and
+ * <code>setLength()</code> methods allow to handle arrays much like array lists. This can be very
+ * useful when efficiency (or syntactic simplicity) reasons make array lists unsuitable.
+ *
+ * <P>
+ * Note that {@link it.unimi.dsi.fastutil.io.BinIO} and {@link it.unimi.dsi.fastutil.io.TextIO}
+ * contain several methods make it possible to load and save arrays of primitive types as sequences
+ * of elements in {@link java.io.DataInput} format (i.e., not as objects) or as sequences of lines
+ * of text.
+ *
+ * @see java.util.Arrays
+ */
+public class BooleanArrays {
```

Comment:
You could use https://docs.oracle.com/javase/7/docs/api/java/util/Arrays.html#fill(boolean[],%20boolean) instead? Then you don't need this class.